internal/transaction/transaction.repository.go

```diff
@@ -0,0 +1,67 @@
+package transaction
+
+import (
+	"Pelter_backend/internal/entity"
+	"fmt"
+
+	"context"
+
+	"gorm.io/gorm"
+)
+
+type (
+	transactionRepository struct {
+		Db *gorm.DB
+	}
+
+	TransactionRepository interface {
+		CreateTransaction(ctx context.Context, txn *entity.Transaction) error
+		GetTransactions(ctx context.Context) ([]*entity.Transaction, error)
+		FindByUserID(ctx context.Context, userID uint) ([]*entity.Transaction, error)
+		FindByTransactionID(ctx context.Context, ID uint) (*entity.Transaction, error)
+	}
+)
+
+func (r *transactionRepository) transactionTable(pctx context.Context) *gorm.DB {
+	return r.Db.Table("transactions").WithContext(pctx)
+}
+
+func NewTransactionRepository(db *gorm.DB) TransactionRepository {
+	return &transactionRepository{
+		Db: db,
+	}
+}
+
+func (r *transactionRepository) CreateTransaction(pctx context.Context, txn *entity.Transaction) error {
+	return r.transactionTable(pctx).Create(txn).Error
+}
+
+func (r *transactionRepository) GetTransactions(pctx context.Context) ([]*entity.Transaction, error) {
+	var txns []*entity.Transaction
```

Comment:
Here a variable is declared as a pointer, but in FindByTransactionId, a regular variable is used
Choose one approach have some consistency

Reply:
the commit 0bdb69782875a0f7e1a2f41c87f294d782d5588d solved the issue...